.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,30 @@
+name: Integration Tests
+
+on:
+    - pull_request
+    - push
+
+jobs:
+  dispatch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the branch name
+        id: get_branch
+        run: echo ::set-output name=branch::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}})
+
+      - name: Get the originating repository
+        id: get_repo
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo ::set-output name=repo::${{ github.event.pull_request.head.repo.full_name }}
+          else
+            echo ::set-output name=repo::${{ github.repository }}
+          fi
+
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@ff45666b9427631e3450c54a1bcbee4d9ff4d7c0 # v3.0.0
+        with:
+          token: "${{ secrets.PAT_INTEGRATION_TESTS}}"
```

Comment:
I am not sure, but can't you use an organization fine-grained token?

Reply:
I think he said it was in this thread: https://github.com/expressjs/express/pull/5492#discussion_r1493812570

Reply:
Yes, we can use a fine-grained one.